Expérience : utiliser Stimulus (avec aussi un peu de TypeScript)

davidbgk opened the pull request:
- [ ] rendre le feedback-endpoint-value configurable (pas en dur)
- [ ] formulaire de drapeau lorsqu'on passe sur les pages du questionnaire

davidbgk committed:
Tech: expérimentation de passage à Stimulus

Pour le formulaire de feedback global.

Avec @rik

davidbgk commented:
Note en voulant ajouter le `switch` depuis la soumission du `feedback` vers `thankyou`, dans la précédente version on affichait également `partage`, j'hésite entre dupliquer le markup ou adapter le code (ou retirer le partage :p).

davidbgk committed:
Tech: suite de l'exploration Stimulus, avec tests

Avec @rik

davidbgk committed:
Tech: réutilisation de feedback.envoieLesRemarques

davidbgk committed:
Tech: retrait d'une virgule inutile

davidbgk committed:
Tech: utilisation de `navigator.share` via Stimulus

davidbgk committed:
Tech: remplace l'ancien feedback par Stimulus

davidbgk committed:
Tech: utilisation de Stimulus pour feedback inline

Il reste quelques TODOs pour pairer/trinomer.

ronnix committed:
Tech: déplace une fonction

ronnix committed:
Tech: supprime bindFeedback (code mort)

ronnix committed:
Tech: supprime la classe feedback-component

On va essayer de n’utiliser les classes que pour les styles (CSS).

ronnix committed:
Tech: passe prettier sur le HTML d’un composant de feedback

ronnix committed:
Ordre des attributs HTML pour Stimulus

On se dit que l’ordre suivant serait pas mal :
- attributs sans data
- attributs data de Stimulus
    - data-controller
    - data-xxx-value
    - data-xxx-target
    - data-action
    - data-xxx-param
- attributs data non Stimulus

ronnix committed:
Tech: supprime une classe CSS non utilisée

ronnix committed:
Déplace la responsabilité du endpoint au HTML

ronnix committed:
Tech: sort le calcul des métadonnées globales du contrôleur

ronnix committed:
Il faudra penser à passer la question lorsque nécessaire !

ronnix committed:
Simplifie le contrôleur de partage

ronnix committed:
Inverse la logique du contrôleur de partage

ronnix committed:
Déplace la logique à l’instanciation du contrôleur

ronnix commented:
Mauvaise version de prettier chez moi ? Je corrige le commit concerné et je `push -f`.

davidbgk committed:
Tech: le feedback devient un formulaire avec inputs

Sémantiquement, il est plus propre que le choix de réponse soit stocké dans le HTML grâce à un input de type `radio`. Cela permet aussi d'y accéder via `FormData`.

Notes:

* `spreadReponse` pourrait être un controlleur dédié.
* la div `feedback-controls` pourrait être un `fieldset` ?
* le HTML pourrait être extrait du Python

davidbgk committed:
Tech: extraction du HTML contenu dans le Python

davidbgk committed:
Tech: mise à jour des tests de rendu markdown

davidbgk committed:
Tech: drapeau sur une page en Stimulus

davidbgk committed:
Passage de la question lors de l'envoi des réponses

davidbgk committed:
Retrait du pageChanged pour le feedback drapeau

ronnix committed:
Modifie les tests de l’API de feedback pour coller au frontend

davidbgk committed:
Amélioration de la fonction de clonage 🐑

rik committed:
Configuration minimale pour débuter Typescript

# Conflicts:
#	Makefile
#	package.json

davidbgk committed:
Passage du share controller en TS

On n’utilise pas le ActionEvent de Stimulus car on n’a pas besoin des targets.

On a activé allowJs pour permettre de faire une transition JS => TS sans erreurs sur les fichiers JS pas encore convertis (notamment `affichage.js`)

En activant cette option, on a découvert que les types de mocha-params dépendaient des types de mocha qui n’était pas présents. On les installe à la main.

davidbgk committed:
Conversion du Plausible controller

Où l’on apprend à déclarer ce qu’il y a dans l’espace de nom global (window.app).

On introduit aussi la notion de unknown pour aider à remonter les incohérences de types au cours de la migration.

davidbgk committed:
Conversion du feedback controller

On crée notre propre classe de Controller Stimulus qui permet de spécifier la nature de `this.element` lors de la définition du controleur.

On apprend à déclarer les accesseurs des values et target(s) qui sont dynamiques.

On passe un moment à essayer d’ajouter le polyfill fromEntries et à le faire reconnaitre par ts-lint, lorsqu’on l’importe depuis core-js (`core-js/proposals/object-from-entries`) ça ne semble pas être reconnu.

On finit par abandonner son usage qui n’était pas critique (pertinent ?) et par accéder aux données du FormData directement.

Au passage, on apprend que l’on peut utiliser `"lib": ["es5", "es2019.object", "dom"],` dans la tsconfig si on ajoute des polyfills particuliers afin que le linter les reconnaisse.

rik commented:
[Pull request ouverte pour améliorer Stimulus](https://github.com/hotwired/stimulus/pull/529).
